crates/redis/src/lib.rs

```diff
@@ -107,15 +110,29 @@ impl TriggerExecutor for RedisTrigger {
 impl RedisTrigger {
     // Handle the message.
     async fn handle(&self, msg: redis::Msg) -> Result<()> {
-        let channel = msg.get_channel_name();
+        let channel = msg.get_channel_name().to_owned();
         tracing::info!("Received message on channel {:?}", channel);
 
-        if let Some(component_id) = self.channel_components.get(channel) {
-            tracing::trace!("Executing Redis component {component_id:?}");
-            let executor = SpinRedisExecutor;
-            executor
-                .execute(&self.engine, component_id, channel, msg.get_payload_bytes())
-                .await?
+        if let Some(component_ids) = self.channel_components.get(&channel) {
+            let futures: Vec<_> = component_ids
+                .iter()
+                .map(|id| {
+                    tracing::trace!("Executing Redis component {id:?}");
+                    SpinRedisExecutor.execute(&self.engine, id, &channel, msg.get_payload_bytes())
+                })
+                .collect();
```

Comment:
I think turning this into a `Vec` isn't necessary as `join_all` works on `intoIterator`.